packages/ui/src/components/Node/ItemCollection.ts

```diff
@@ -1,26 +1,83 @@
 import { ItemValue } from '@data-story/core/*'
 
+type JSONValue = string | number | boolean | {[key: string]: JSONValue} | JSONValue[];
+
 export class ItemCollection {
-  constructor(public items: ItemValue[]) {}
-
-  toTable() {
-    let headers: string[] = []
-    let rows: ItemValue[] = []
-
-    // Extract headers and remove duplicates
-    headers = Array.from(new Set(this.items
-      .map((item) => Object.keys(item))
-      .flat()));
-
-    // Extract rows
-    rows = this.items.map((item) => {
-      const row: ItemValue[] = []
-      headers.forEach((header) => {
-        row.push(item[header])
-      })
-      return row
-    })
-
-    return { headers, rows }
+  constructor(public items: ItemValue[]) {
+  }
+
+  private serializeArrayContent(arrayContent: unknown[]): string {
+    return JSON.stringify(arrayContent).replace(/\n/g, '');
   }
-}
\ No newline at end of file
+
+  public toTable() {
```

Comment:
Public is infered by default, so we can leave this out. Do keep public methods above private ones

Reply:
🆗